app/src/validators/common.js

```diff
@@ -83,8 +83,30 @@ const scheme = {
   string: oneOrMany(Joi.string().max(255)),
 
   pagination: (sortList) => ({
-    page: Joi.number().min(1),
-    limit: Joi.number().min(0),
+    page: Joi.alternatives()
+      .conditional('limit', {
+        not: true,
+        then: Joi
+          .number()
+          .min(0)
+          .required()
+          .messages({
+            'any.required': '`Must specify page number`',
+          }),
+        otherwise: Joi.number().min(0)
+      }),
+    limit: Joi.alternatives()
+      .conditional('page', {
+        not: true,
+        then: Joi
+          .number()
+          .min(1)
+          .required()
+          .messages({
+            'any.required': '`Must specify page limit`',
+          }),
+        otherwise: Joi.number().min(1)
+      }),
     sort: Joi.string().valid(...sortList),
```

Comment:
i dont think this needs the spread operator thing `...`.
 also sort and order dont need to be defined in the custom pagination schema.. but the way you have it is fine.